docs/data/core-blocks.md

```diff
@@ -0,0 +1,86 @@
+# **core/blocks**: Block Types Data
+
+## Selectors 
+
+### getBlockType
+
+Returns a block type by name.
+
+*Parameters*
+
+ * state: Data state.
+ * name: Block type name.
+
+
+### getCategories
+
+Returns all the available categories.
+
+*Parameters*
+
+ * state: Data state.
+
+*Returns*
+
+Categories list.
+
+### getDefaultBlockName
+
+Returns the name of the default block name.
+
+*Parameters*
+
+ * state: Data state.
+
+*Returns*
+
+Default block name.
+
+### getFallbackBlockName
+
+Returns the name of the fallback block name.
+
+*Parameters*
+
+ * state: Data state.
+
+*Returns*
+
+Fallback block name.
+
+## Actions
+
+### addBlockTypes
+
+Returns an action object used in signalling that block types have been added.
```

Comment:
I think the basic redux half-explainations here (an action function returns an action object used by `dispatch()`) aren't helpful and might even be confusing. Instead I'd structure this section like:

```markdown
## Actions

These functions all return [action objects](DOC ON REDUX ACTION OBJECT.html) that are passed to a store's `dispatch()` function to dispatch a redux action.

### addBlockTypes

Signal that a block type, or types, have been added.
```

Reply:
I think it's important to see these docs as a reference API. IMO, These docs shouldn't explain the concepts (actions, redux...) but serve as a reference while the README of the data module (which is already in a good shape) should explain the concepts and link to the reference.

Reply:
But I agree that we can reword the descriptions of the action creators.

Reply:
But it's also misleading to say they signal something. On their own, the functions do not signal anything, and in fact it is accurate to say they merely return an object.

Reply:
Sure. Saying that once above is better and makes the intent of all of these functions clearer in the docs to me.

But it sounds like both you and @youknowriad think they should serve as more of a clear-cut functional reference, so in that case I am out-voted! 🗳